### PR TITLE
App Config Provider Base

### PR DESCRIPTION
**Pull request by mrm9084**
# Description

Moves shared provider code to it's own file, and made a base class for the provider

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by albertofori — APPROVED** [review]

File: sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_client_manager.py [hunk 1]
```diff
@@ -135,11 +135,11 @@ def _check_configuration_setting(
 
     @distributed_trace
     def load_configuration_settings(
-        self, setting_selectors: List[SettingSelector], refresh_on_keys: Dict[Tuple[str, str], str], **kwargs
+        self, selects: List[SettingSelector], refresh_on: Mapping[Tuple[str, str], Optional[str]], **kwargs
```
Comment (albertofori) [inline sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_client_manager.py:138]: I see we previously renamed this to `setting_selectors`, is the intention to revert this change?

Comment (mrm9084) [reply sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_client_manager.py:138]: This rename was the start of going down the wrong path of trying to fix an issue. If you look at `_azureappconfigurationproviderbase` it now has a method `sdk_allowed_kwargs` this is the suggested solution to this issue from the SDK team. TLDR, `kwargs` has a key in in named selects that was colliding with the property here.

**Review by albertofori — APPROVED** [review]

File: sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationproviderbase.py [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+        key, label = setting  # type:ignore
+    except IndexError:
+        key = setting
+        label = EMPTY_LABEL
+    if "*" in key or "*" in label:
+        raise ValueError("Wildcard key or label filters are not supported for refresh.")
+    return key, label
+
+
+def sdk_allowed_kwargs(kwargs):
```
Comment (avanigupta) [inline sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationproviderbase.py:162]: Is this new?

Comment (mrm9084) [reply sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationproviderbase.py:162]: Yes, the SDK throws an error if you pass an unknown kwarg into it. The SDK team suggested adding it. This came up in this PR as after I made the update the items used in the base class were not being popped from the kwargs any more due to the redesign and the SDK team suggested this as a solution.

Comment (avanigupta) [reply sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationproviderbase.py:162]: can sdk team provide this list or supported args? Instead of us having to manage it long term

Comment (mrm9084) [reply sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationproviderbase.py:162]: No, I had to get the list from the readme doc of azure core

Comment (avanigupta) [reply sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationproviderbase.py:162]: Discussed offline. SDK team suggested this approach for Python as they dont support any other option right now. We can revisit this when a better option is available.

**Review by avanigupta — APPROVED** [review]
